### ISSUE TITLE
Step "Publish and install Microsoft extensions" needs code to update all apps with one command

### ISSUE THREAD

**Issue by erolkunman** [MicrosoftDocs/dynamics365smb-devitpro-pb#1893]
I have manual labor issue with step: "Publish and install Microsoft extensions" and instruction "Complete these steps for each Microsoft extension that you want to upgrade."

Currently Microsoft has 20 extensions and with every version there will be more.
It's a LOT of manual work to update all these extensions individually ... surely Microsoft engineers can make PowerShell command/function, that goes through all extensions in installation DVD folder Applications and checks if extension is installed. If extension is installed, then script should upgrade the app and finally remove the old version.

Thank You in advance for this code/function.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 86cc4b47-eb79-1a2f-27c4-692bd92e4a9b
* Version Independent ID: 23adc9aa-52b1-7329-252a-67c834dd4cb0
* Content: [Install a version 18 update - Business Central](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/upgrade/upgrading-cumulative-update-v18)
* Content Source: [dev-itpro/upgrade/upgrading-cumulative-update-v18.md](https://github.com/MicrosoftDocs/dynamics365smb-devitpro-pb/blob/main/dev-itpro/upgrade/upgrading-cumulative-update-v18.md)
* Service: **dynamics365-business-central**
* GitHub Login: @jswymer
* Microsoft Alias: **jswymer**

**Comment by git-no** [comment]
This already possible by using Powershell and the current NAV/BC-Commands. Write a script by your own. I do not see why a comfort function should be provided by Microsoft.
- Read the extension files you want to publish (get-item).
- Publish, update/install.
- The whole BC update/upgrade process can run via powershell. There are examples on the DVD.

**Comment by erolkunman** [comment]
This is absurd.

Currently Business Central OnPrem is the ONLY software I have to manually upgrade (meaning every month I manually download new "DVD" installer, then uninstall + reinstall + upgrade etc). All others are upgraded in the background with none ore minimal user involvement.

**What Microsoft should do is make a proper modern installer, that handles upgrade as expected in 21st century.**
Current upgrade process is too complicated even for system administrators.
And that's why customers are stuck on old+buggy BC versions.

>The whole BC update/upgrade process can run via powershell. There are examples on the DVD.
WHERE? In folder WindowsPowerShellScripts\Upgrade there are HISTORIC (NAV70) upgrade scripts....it's like utter nonsense even to include these with BC18 DVD. _And this is year 2021 ... DVD is DEAD in my opinion anyway._

>Read the extension files you want to publish (get-items).
I'm a functional consultant and not a powershell developer. I honestly tried but did not succeed.

> I do not see why a comfort function should be provided by Microsoft.
Comfort? All of the upgrade process is **PAIN** not comfort.
Microsoft should provide this function because then ALL USERS that want to upgrade, can then use this function and don't have to invent a bicycle over and over again.

I have made numerous suggestions to this upgrade guide during the past few years in order to make Microsoft upgrade instructions at least work. Previously they were so bad, that upgrade could not be completed.

@git-no if You have working no-hassle upgrade script please share it.

**Comment by git-no** [comment]
@erolkunman I can understand that the development and DevOps part seems to be complicated as a none technician. In fact also for NAV developers it is not easy, the last years brought a lot of new methods and tools to handle.
I do not know if you work with Docker containers but this would be the state of the art and proposed way nowadays (look at NAVContainerHelper). We working daily with different BC/NAV versions to support customers and it works, after a preparation period to build up the process, quite well.
Building up automation in development, support and adminitration and even the BC upgrade process took effort. for sure the BC upgrade module we wrote will need consulting, introduction and support. Sorry, but I hope you do understand this is nothing we can share without compensation. Hope you find your pain relieves.

**Comment by SusanneWindfeldPedersen** [comment]
Thanks for your feedback. This repo is used for documentation feedback. Please go to [Ideas](https://experience.dynamics.com/ideas/categories/?forum=e288ef32-82ed-e611-8101-5065f38b21f1&forumName=Dynamics%20365%20Business%20Central) to suggest a product feature. Thanks

**Comment by erolkunman** [comment]
@SusanneWindfeldPedersen do You really think I would be wasting my time here if I don't know the difference between documentation and product feature?

If then what do You call current powershell example:
Publish-NAVApp -ServerInstance $BcServerInstance -Path $ExtPath

So let me rephrase:
Surely Microsoft engineers can make public an EXAMPLE PowerShell script, that goes through all extensions in installation DVD folder Applications and checks if extension is installed. If extension is installed, then script should upgrade the app and finally remove the old version.

Thanks for nothing I suppose.
